Service/ReplicaSettingsHandler.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Service;
+
+use Algolia\AlgoliaSearch\Api\Data\IndexOptionsInterface;
+use Algolia\AlgoliaSearch\Exceptions\AlgoliaException;
+use Algolia\AlgoliaSearch\Helper\ConfigHelper;
+use Magento\Framework\Exception\NoSuchEntityException;
+
+/**
+ *  This class will proxy the settings save operations and forward to replicas based on user configuration
+ *  excluding attributes which should never be forwarded
+ */
+class ReplicaSettingsHandler
+{
+    /**
+     * As replicas are used for sorting we do not want to override these replicas specific configurations
+     */
+    protected const FORWARDING_EXCLUDED_ATTRIBUTES = [
+        'customRanking',
+        'ranking'
+    ];
+
+    public function __construct(
+        protected AlgoliaConnector $connector,
+        protected ConfigHelper     $config,
+    ) {}
+
+    /**
+     * @throws NoSuchEntityException
+     * @throws AlgoliaException
+     */
+    public function setSettings(
+        IndexOptionsInterface $indexOptions,
+        array $indexSettings,
+        string $mergeSettingsFrom = ''
+    ): void
+    {
+        if ($this->config->shouldForwardPrimaryIndexSettingsToReplicas($indexOptions->getStoreId())) {
+            [$forward, $noForward] = $this->splitSettings($indexSettings);
+            if ($forward) {
+                $this->connector->setSettings(
+                    $indexOptions,
+                    $forward,
+                    true,
+                    false
+                );
+            }
+            if ($noForward) {
+                $this->connector->setSettings(
+                    $indexOptions,
+                    $noForward,
+                    false,
+                    true,
+                    $mergeSettingsFrom
+                );
+            }
+        } else {
```

Comment:
I think this method would be simpler if the `if` condition was inverted and if this was an early return.

Reply:
Yeah good callout. Will tweak that. Thanks!